tests/011-test-autoextend-by-amount.sh

```diff
@@ -0,0 +1,120 @@
+source $SRCDIR/libtest.sh
+
+cleanup_lvm_thin_device() {
+  local device_mount=$1
+  local thin_device=$2
+
+  umount $device_mount
+  rm -rf $device_mount
+  lvremove -f $thin_device >> $LOGS 2<&1
+}
+  
+# Test if the docker-pool is auto-extending, and by the
+# right amount, which by default is 20% of the LV size.
+test_auto_extend_by_amount() {
+
+  local devs=$TEST_DEVS dev
+  local test_status
+  local testname=`basename "$0"`
+  local vg_name
+  local vg_name="dss-test-foo"
+  local pool_name="docker-pool"
+  local test_status
+  local threshold=60
+
+  # Error out if any pre-existing volume group vg named dss-test-foo
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      echo "ERROR: $testname: Volume group $vg_name already exists."
+      return 1
+    fi
+  done
+  
+  # Error out if any pre-existing directory WORKDIR/test-thin-dev 
+  if [ -d $WORKDIR/test-thin-dev ]; then
+    echo "ERROR: $testname: Directory $WORKDIR/test-thin-dev already exists."
+    return 1
+  fi
+
+  local thin_dev_mount=$WORKDIR/test-thin-dev
+  
+  # Create config file
+  clean_config_files
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+EOF
+
+  test_status=1
+  # Run docker-storage-setup
+  $DSSBIN >> $LOGS 2>&1
+ 
+  # Make sure volume group $VG got created.
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      test_status=0
+      break
+    fi
+  done  
+  
+  # error out if vg wasn't created
+  if [ "$test_status" -ne 0 ]; then
+    cleanup $vg_name "$devs"
+    return $test_status
+  fi
+  
+  test_status=1
+ 
+  # Simulate writing data to the thin volume and seeing if it auto-extends
+  lvcreate -n docker-thin-device -V 1T --thinpool $vg_name/$pool_name >> $LOGS 2<&1
+  mkfs.ext4 /dev/$vg_name/docker-thin-device >> $LOGS 2<&1
+  mkdir $WORKDIR/test-thin-dev >> $LOGS 2<&1
+  mount /dev/$vg_name/docker-thin-device $WORKDIR/test-thin-dev >> $LOGS 2<&1
```

Comment:
You don't have to really mount device. you can do block writes using "dd" directly on block device.